core/logger/logger.go

```diff
@@ -192,42 +196,76 @@ func NewLogger() Logger {
 		parseErrs = append(parseErrs, invalid)
 	}
 
-	c.ToDisk, invalid = envvar.LogToDisk.ParseBool()
+	var fileMaxSize utils.FileSize
+	fileMaxSize, invalid = envvar.LogFileMaxSize.ParseFileSize()
+	c.DiskMaxSizeBeforeRotate = int(fileMaxSize)
 	if invalid != "" {
 		parseErrs = append(parseErrs, invalid)
+	} else {
+		c.ToDisk = c.DiskMaxSizeBeforeRotate > 0
+	}
+
+	if c.ToDisk {
```

Comment:
I used old references in code for this. But to avoid further confusion, i've had removed them entirely. Now everything should point to `DebugLogsToDisk` instead.

Reply:
A major refactor happened here in which the conditional on whether or not to store logs in disk is simpler, and removes the confusion from the older version of the feature.